eslint.config.js

```diff
@@ -0,0 +1,77 @@
+// @ts-check
+
+// Todo: Use ES module
+const eslint = require('@eslint/js');
+const tseslint = require('typescript-eslint');
+const reactRefresh = require('eslint-plugin-react-refresh');
+
+const FlatCompat = require('@eslint/eslintrc').FlatCompat;
+const compat = new FlatCompat({
+  baseDirectory: `${__dirname}`
+});
+
+module.exports = tseslint.config(
+  { ignores: ['eslint.config.js'] },
+  // eslint.configs.recommended,
+  ...tseslint.configs.recommended,
+  // TODO: Enable when ready
+  // {
+  //   plugins: {
+  //     'react-refresh': reactRefresh
+  //   },
+  //   rules: {
+  //     'react-refresh/only-export-components': 'warn'
+  //   }
+  // },
```

Comment:
Should I resolve the warnings in this PR @chownces? Feels a bit out of scope though, I think better to resolve all warnings in one go in a separate PR.

Reply:
Done in #2990